istioctl/pkg/writer/envoy/configdump/listener.go

```diff
@@ -102,18 +107,166 @@ func (c *ConfigWriter) PrintListenerSummary(filter ListenerFilter) error {
+	istioHTTPPlaintext = []string{"istio", "istio-http/1.0", "istio-http/1.1", "istio-h2"}
+	httpTLS            = []string{"http/1.0", "http/1.1", "h2c", "istio-http/1.0", "istio-http/1.1", "istio-h2"}
+	tcpTLS             = []string{"istio-peer-exchange", "istio"}
+
+	protDescrs = map[string][]string{
+		"App: HTTP TLS":         httpTLS,
+		"App: Istio HTTP Plain": istioHTTPPlaintext,
+		"App: TCP TLS":          tcpTLS,
+		"App: HTTP":             plaintextHTTPALPNs,
+	}
+)
+
+func retrieveListenerMatches(l *listener.Listener) []filterchain {
+	resp := []filterchain{}
+	for _, filterChain := range l.GetFilterChains() {
+		match := filterChain.FilterChainMatch
+		if match == nil {
+			match = &listener.FilterChainMatch{}
+		}
+		// filterChaince also has SuffixLen, SourceType, SourcePrefixRanges which are not rendered.
+
+		descrs := []string{}
+		if len(match.ServerNames) > 0 {
+			descrs = append(descrs, fmt.Sprintf("SNI: %s", strings.Join(match.ServerNames, ",")))
+		}
+		if len(match.TransportProtocol) > 0 {
+			descrs = append(descrs, fmt.Sprintf("Trans: %s", match.TransportProtocol))
+		}
+
+		if len(match.ApplicationProtocols) > 0 {
+			found := false
+			for protDescr, protocols := range protDescrs {
+				if reflect.DeepEqual(match.ApplicationProtocols, protocols) {
+					found = true
+					descrs = append(descrs, protDescr)
+					break
+				}
+			}
+			if !found {
+				descrs = append(descrs, fmt.Sprintf("App: %s", strings.Join(match.ApplicationProtocols, ",")))
+			}
+		}
+
+		port := ""
+		if match.DestinationPort != nil {
+			port = fmt.Sprintf(":%d", match.DestinationPort.GetValue())
+		}
+		if match.AddressSuffix != "" {
+			descrs = append(descrs, fmt.Sprintf("Addr: %s%s", match.AddressSuffix, port))
+		}
+		if len(match.PrefixRanges) > 0 {
+			pf := []string{}
+			for _, p := range match.PrefixRanges {
+				pf = append(pf, fmt.Sprintf("%s/%d", p.AddressPrefix, p.GetPrefixLen().GetValue()))
+			}
+			descrs = append(descrs, fmt.Sprintf("Addr: %s%s", strings.Join(pf, ","), port))
+		}
+		if len(descrs) == 0 {
+			descrs = []string{"ALL"}
+		}
+		fc := filterchain{
+			destination: getFilterType(filterChain.GetFilters()),
+			match:       strings.Join(descrs, "; "),
+		}
+		resp = append(resp, fc)
+	}
+	return resp
+}
+
+func getFilterType(filters []*listener.Filter) string {
+	for _, filter := range filters {
+		if filter.Name == HTTPListener {
+
+			httpProxy := &httpConn.HttpConnectionManager{}
+			// Allow Unmarshal to work even if Envoy and istioctl are different
+			filter.GetTypedConfig().TypeUrl = "type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager"
+			err := ptypes.UnmarshalAny(filter.GetTypedConfig(), httpProxy)
+			if err != nil {
+				return err.Error()
+			}
+			if httpProxy.GetRouteConfig() != nil {
+				return "Inline Route"
+			}
+			if httpProxy.GetRds().GetRouteConfigName() != "" {
+				return httpProxy.GetRds().GetRouteConfigName()
```

Comment:
maybe make this `"Route: 80"`? and for clusters, `"Cluster: outbound||blah"`?

Reply:
@howardjohn Added "Route:" and "Cluster"; also added some Match info, e.g. `Inline Route: /healthz/ready*`